browser/ai_chat/ai_chat_api_request.cc

```diff
@@ -0,0 +1,115 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ai_chat/ai_chat_api_request.h"
+
+#include <base/containers/flat_map.h>
+
+#include <utility>
+
+#include "base/json/json_writer.h"
+#include "brave/browser/ai_chat/buildflags.h"
+#include "brave/browser/ai_chat/common/constants.h"
+#include "brave/components/constants/brave_services_key.h"
+#include "net/traffic_annotation/network_traffic_annotation.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "url/gurl.h"
+
+namespace {
+
+net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
+  return net::DefineNetworkTrafficAnnotation("ai_chat", R"(
+      semantics {
+        sender: "AI Chat"
+        description:
+          "This is used to communicate with our partner API"
+          "on behalf of the user interacting with the ChatUI."
+        trigger:
+          "Triggered by user sending a prompt."
+        data:
+          "Will generate a text completion that attempts to match whatever context or pattern the user gave it"
+        destination: WEBSITE
+      }
+      policy {
+        cookies_allowed: NO
+        policy_exception_justification:
+          "Not implemented."
+      }
+    )");
+}
+
+GURL GetURLWithPath(const std::string& host, const std::string& path) {
+  return GURL(std::string(url::kHttpsScheme) + "://" + host).Resolve(path);
+}
+
+std::string CreateJSONRequestBody(base::ValueView node) {
+  std::string json;
+  base::JSONWriter::Write(node, &json);
+  return json;
+}
+
+}  // namespace
+
+AIChatAPIRequest::AIChatAPIRequest(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : api_request_helper_(GetNetworkTrafficAnnotationTag(),
+                          url_loader_factory) {}
+
+AIChatAPIRequest::~AIChatAPIRequest() = default;
+
+void AIChatAPIRequest::QueryPrompt(ResponseCallback callback,
+                                   const std::string& prompt) {
+  auto internal_callback =
+      base::BindOnce(&AIChatAPIRequest::OnGetResponse,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  base::Value::Dict dict;
+  base::Value::List stop_sequences;
+  stop_sequences.Append("\n\nHuman:");
+
+  dict.Set("prompt", prompt);
```

Comment:
Is the meaning of these parameters documented anywhere?

Reply:
If we can, it would be nice to add a few comments here to the non-self-explanatory params (e.g. `temperature`).

Reply:
Sent doc on slack

Reply:
Did you add the comments?